cmake/modules/OmrTest.cmake

```
@@ -0,0 +1,49 @@
+###############################################################################
+# Copyright (c) 2021, 2021 IBM Corp. and others
+#
+# This program and the accompanying materials are made available under
+# the terms of the Eclipse Public License 2.0 which accompanies this
+# distribution and is available at http://eclipse.org/legal/epl-2.0
+# or the Apache License, Version 2.0 which accompanies this distribution
+# and is available at https://www.apache.org/licenses/LICENSE-2.0.
+#
+# This Source Code may also be made available under the following Secondary
+# Licenses when the conditions for such availability set forth in the
+# Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+# version 2 with the GNU Classpath Exception [1] and GNU General Public
+# License, version 2 with the OpenJDK Assembly Exception [2].
+#
+# [1] https://www.gnu.org/software/classpath/license.html
+# [2] http://openjdk.java.net/legal/assembly-exception.html
+#
+# SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+###############################################################################
+
+if(OMR_TEST_)
+	return()
+endif()
+set(OMR_TEST_ 1)
+
+if(NOT OMR_TEST_LAUNCHER)
+	if(OMR_EXE_LAUNCHER)
+		set(OMR_TEST_LAUNCHER "${OMR_EXE_LAUNCHER}")
+	endif()
+endif()
+
+function(omr_add_test)
+	cmake_parse_arguments(
+        opt    # prefix of output variables
+        ""     # list of names of the boolean arguments (only defined ones will be true)
+        "NAME;COMMAND" # list of names of mono-valued arguments
+        "" # list of names of multi-valued arguments (output variables are lists)
+        ${ARGN}# arguments of the function to parse, here we take the all original ones
+    )
```

The comments repeat what the [documentation already describes](https://cmake.org/cmake/help/latest/command/cmake_parse_arguments.html). I don't think they're necessary.

```suggestion
	cmake_parse_arguments(opt "" "NAME;COMMAND" "" ${ARGN})
```